lib/api/apiUtils/bucket/models/ReplicationConfiguration.js

```diff
@@ -0,0 +1,331 @@
+        }
+        return obj;
+    }
+
+    /**
+     * Check that the `Role` property of the configuration is valid
+     * @return {undefined}
+     */
+    _checkRole() {
+        const Role = this._parsedXML.ReplicationConfiguration.Role;
+        if (!Role) {
+            return errors.MalformedXML;
+        }
+        // TODO: Update to validate role priveleges after implemented in Vault.
+        // Role should be an IAM user name.
+        const arr = Role[0].split(':');
+        const isValidARN = arr.length === 7 ||
+            (arr.length === 6 && arr[5].split('/').length === 2);
+        if (!isValidARN) {
+            return errors.InvalidArgument.customizeDescription(
+                'Invalid Role specified in replication configuration');
+        }
+        this.replicationConfiguration.role = Role[0];
+        return undefined;
+    }
+
+    /**
+     * Check that the `Rules` property array is valid
+     * @return {undefined}
+     */
+    _checkRules() {
+        // Note that the XML uses 'Rule' while the config object uses 'Rules'.
+        const { Rule } = this._parsedXML.ReplicationConfiguration;
+        if (!Rule || Rule.length < 1) {
+            return errors.MalformedXML;
+        }
+        if (Rule.length > 1000) {
+            return errors.InvalidRequest.customizeDescription(
+                'Number of defined replication rules cannot exceed 1000');
+        }
+        const err = this._checkEachRule(Rule);
+        if (err) {
+            return err;
+        }
+        this.replicationConfiguration.rules = this._rules;
+        return undefined;
+    }
+
+    /**
+     * Check that each rule in the `Rules` property array is valid
+     * @param {array} rules - The rule array from this._parsedXML
+     * @return {undefined}
+     */
+    _checkEachRule(rules) {
+        const rulesArr = [];
+        for (let i = 0; i < rules.length; i++) {
+            const err =
+                this._checkStatus(rules[i]) || this._checkPrefix(rules[i]) ||
+                this._checkID(rules[i]) || this._checkDestination(rules[i]);
+            if (err) {
+                return err;
+            }
+            rulesArr.push(this._getRuleObject(rules[i]));
+        }
+        this._rules = rulesArr;
+        return undefined;
+    }
+
+    /**
+     * Check that the `Status` property is valid
+     * @param {object} rule - The rule object from this._parsedXML
+     * @return {undefined}
+     */
+    _checkStatus(rule) {
+        const status = rule.Status && rule.Status[0];
+        if (!status || !['Enabled', 'Disabled'].includes(status)) {
+            return errors.MalformedXML;
+        }
+        return undefined;
+    }
+
+    /**
+     * Check that the `Prefix` property is valid
+     * @param {object} rule - The rule object from this._parsedXML
+     * @return {undefined}
+     */
+    _checkPrefix(rule) {
+        const prefix = rule.Prefix && (rule.Prefix[0] || rule.Prefix[0] === '');
```

Comment:
Good catch. I've updated this and added some comments concerning the handling of an empty string prefix.